Assets/Scripts/Section3/PresentPainting.cs

```diff
@@ -30,19 +47,47 @@ protected override void OnUserInteract()
         else
         {
             AudioPlayer.Play(Globals.VoiceLines.Section3.NOT_ALL_FILTERS, Globals.Tags.DIALOGUE_SOURCE);
-        }
+        }*/
         
     }
 
     void OnCanvasChange()
     {
-        Color color = GlobalState.GetVar<Color>(Globals.Vars.DAD_PUZZLE_2_FINAL_LIGHT_COLOR);
+        int filtersLit = getFilterLights();
+        if (filtersLit == FILTER_COUNT)
+        {
+            AudioPlayer.Play(Globals.VoiceLines.Section3.ALL_FILTERS, Globals.Tags.DIALOGUE_SOURCE, false);
+            painting.SetActive(true);
+        }
+        /*Color color = GlobalState.GetVar<Color>(Globals.Vars.DAD_PUZZLE_2_FINAL_LIGHT_COLOR);
+        Debug.Log("OnCanvasChange(): color: " + color + " finalColour: " + finalColour);
         if (additiveColorCheck(color, finalColour))
         {
-            AudioPlayer.Play(Globals.VoiceLines.Section3.ALL_FILTERS, Globals.Tags.DIALOGUE_SOURCE);
+            AudioPlayer.Play(Globals.VoiceLines.Section3.ALL_FILTERS, Globals.Tags.DIALOGUE_SOURCE, false);
+            painting.SetActive(true);
+        }*/
+    }
+    
+    // Temporary fix that checks if all filters are passed through with light. Leading to the solution.
+    private int getFilterLights()
```

Comment:
I think we keep this implementation as we're going to be playing around with the light filters anyway.